### PR TITLE
feat: Anchor component changed to data-driven

### PR DESCRIPTION
**Pull request by foryuki**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     Add items prop to make configuration options content driven by data, and support nesting through children.      |
| 🇨🇳 Chinese |  增加 items prop 数据化配置选项内容，支持通过 children 嵌套 。      |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/anchor/Anchor.tsx [hunk 1]
```diff
@@ -61,6 +69,7 @@ export interface AnchorProps {
   targetOffset?: number;
   /** Listening event when scrolling change active link */
   onChange?: (currentActiveLink: string) => void;
+  items: AnchorLinkItemProps[];
```
Comment (abenhamdine) [inline components/anchor/Anchor.tsx:72]: shouldn't it be 
```suggestion
  items?: AnchorLinkItemProps[];
```

**Review by abenhamdine — COMMENTED** [review]
it could also be useful to add `@deprecated` tag on children property

File: components/anchor/demo/basic.tsx [hunk 2]
```diff
@@ -1,17 +1,40 @@
...
-  <Anchor>
-    <Link href="#components-anchor-demo-basic" title="Basic demo" />
-    <Link href="#components-anchor-demo-static" title="Static demo" />
-    <Link href="#api" title="API">
-      <Link href="#anchor-props" title="Anchor Props" />
-      <Link href="#link-props" title="Link Props" />
-    </Link>
-  </Anchor>
+  <Anchor
+    items={[
```
Comment (foryuki) [inline components/anchor/demo/basic.tsx:6]: @MadCcc @zombieJ  demo 是否要替换成这种形式？

Comment (zombieJ) [reply components/anchor/demo/basic.tsx:6]: 嗯，换一下~

Comment (zombieJ) [reply components/anchor/demo/basic.tsx:6]: v5 里就都推荐 data driven 了~

Comment (foryuki) [reply components/anchor/demo/basic.tsx:6]: 好的👌

**Review by MadCcc — APPROVED** [review]

**Pull request merged**